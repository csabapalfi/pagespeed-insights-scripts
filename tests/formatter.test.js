const format = require('../lib/formatter');

describe('format', () => {
  describe('results', () => {
    const result = {
      type: 'result',
      name: 'run 1',
      score: 95,
      FCP: 0.9,
      FMP: 1,
      SI: 1.1,
      FCI: 3.2,
      TTI: 4
    };

    it('are formatted as expected', () => {
      expect(format([result], {}))
        .toMatchSnapshot();
    });
  
    it('heading row added before first result', () => {
      expect(format([result], {first: true}))
        .toMatchSnapshot();
    });

    it('userTimingMark uses default number format precision (2 digits)', () => {
      expect(format([{...result, DPA: 1.25}], {}))
        .toMatchSnapshot();
    });
  });

  describe('stats', () => {
    const statistic = {
      type: 'statistic',
      name: 'median',
      score: 95,
      FCP: 0.9,
      FMP: 1,
      SI: 1.1,
      FCI: 3.2,
      TTI: 4
    };
  
    it('stats are formatted as expected', () => {
      expect(format([statistic]))
        .toMatchSnapshot();
    });

    it('stddev uses custom precision for score (1 digit)', () => {
      expect(format([{...statistic, name: 'stddev'}]))
        .toMatchSnapshot();
    });
  });
});