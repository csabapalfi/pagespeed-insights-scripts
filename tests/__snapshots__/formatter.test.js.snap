// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format results are formatted as expected 1`] = `
Array [
  "run 1 	95	0.9	1.0	1.1	3.2	4.0",
]
`;

exports[`format results heading row added before first result 1`] = `
Array [
  "name  	score	FCP	FMP	SI	FCI	TTI",
  "run 1 	95	0.9	1.0	1.1	3.2	4.0",
]
`;

exports[`format results userTimingMark uses default number format precision (2 digits) 1`] = `
Array [
  "run 1 	95	0.9	1.0	1.1	3.2	4.0	1.25",
]
`;

exports[`format stats stats are formatted as expected 1`] = `
Array [
  "",
  "median	95	0.9	1.0	1.1	3.2	4.0",
]
`;

exports[`format stats stddev uses custom precision for score (1 digit) 1`] = `
Array [
  "",
  "stddev	95.0	0.9	1.0	1.1	3.2	4.0",
]
`;
